app_backend/server.py

```diff
@@ -56,5 +59,76 @@ async def add_friend(resp: Response, uuid: str = None, fuuid: str = None):
     uuid_friends.append(fuuid)
     return {"msg" : "FUUID has been successfully linked to UUID"}
 
+"""
+    UserLogin model keeps track of the user login information
+    username: str
+        The username of the user
+    email: str
+        The email of the user
+    password: str
+        The password of the user
+"""
+class UserLogin(BaseModel):
+    username: str
+    email: str
+    password: str
+
+
+"""
+    getuser implements the route /login/
+    returns message
+"""
+# TODO: Once attached to database will be able to retreive
+# specific user information and to log them in
+@app.get("/login/")
+async def getuser():
+    return {"msg": "This is the login page!"}
+
+
+"""
+    registeruser implements the /register/ route
+    login: UserLogin
+        The UserLogin model that includes the username, email, and password
+    resp: Response
+        The response to send back to the user which contains the status code and body
+    returns Response
+        Response.body: dict
+            Provides any msgs/errs for the request
+        Response.status: int
+            The status code for the request        
+"""
+@app.post("/register/", status_code=status.HTTP_201_CREATED)
+async def registeruser(login: UserLogin, resp: Response):
+
+    username = login.username
+    email = login.email
+    password = login.password
+
+    validUsername = r'^(?=[a-zA-Z0-9._]{8,20}$)(?!.*[_.]{2})[^_.].*[^_.]$'
```

Comment:
Can you explain what these values are, I'm not familiar  @brendacano

Reply:
Will do

Reply:
done, please check if this is what you had in mind